File: leetcode/94/binary_tree_inorder_traversal.cpp
#include <iostream>
#include <vector>

using namespace std;

struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> result;

        if (root == nullptr) {
          return result;
        }

        if (root->left != nullptr) {
          vector<int> leftTree = inorderTraversal(root->left);
          result.insert(result.end(), leftTree.begin(), leftTree.end());
        }

        result.insert(result.end(), 1, root->val);

        if (root->right != nullptr) {
          vector<int> rightTree = inorderTraversal(root->right);
          result.insert(result.end(), rightTree.begin(), rightTree.end());
        }

        return result;
    };
};

File: leetcode/94/binary_tree_inorder_traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if root is None:
            return res

        if root.left is not None:
            res = res + self.inorderTraversal(root.left)

        if root.val is not None:
            res.append(root.val)

        if root.right is not None:
            res = res + self.inorderTraversal(root.right)

        return res
